### ISSUE TITLE
Question: Circular references in Waterline's toObject()

### ISSUE THREAD

**Issue by erictooth** [balderdashy/sails#3076]
Since updating to Sails 0.11, I've had an issue where my app crashes saying "There was an error turning the model into an object" whenever I pull a single model (/api/user/:someid) but (/api/users/) works. I'm just using the built-in blueprints for this.

In Waterline's toObject() function:
```javascript
try {

  // *** new line added below: ***
  // this.object._locals contains circular references and therefore breaks JSON.stringify()
  // if I add this line, my app works correctly
  delete this.object._locals;

  // Stringify/parse the object
  var _obj = JSON.parse(JSON.stringify(this.object));

  return _obj;
}
```

Is there something that I'm doing wrong with my app's configuration, or is there a snippet of code that I can post to help further diagnose the issue?

**Comment by tjwebb** [comment]
Can you share any more info? I don't think this is a very common problem.

**Comment by sailsbot** [comment]
Thanks for posting, @erictooth.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
